db/migrate/20240314165306_migrate_groups_to_filters.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+class MigrateGroupsToFilters < ActiveRecord::Migration[7.0]
+  class BillableMetricFilter < ApplicationRecord
+    include Discard::Model
+    self.discard_column = :deleted_at
+    default_scope -> { kept }
+
+    belongs_to :billable_metric
+  end
+
+  class Group < ApplicationRecord
+    include Discard::Model
+    self.discard_column = :deleted_at
+    default_scope -> { kept }
+
+    belongs_to :billable_metric, -> { with_discarded }
+    belongs_to :parent, -> { with_discarded }, class_name: 'Group', foreign_key: 'parent_group_id', optional: true
+  end
+
+  class Charge < ApplicationRecord
+    has_many :group_properties, dependent: :destroy
+    belongs_to :billable_metric, -> { with_discarded }
+    has_many :filters, dependent: :destroy, class_name: 'ChargeFilter'
+    has_many :filter_values, through: :filters, class_name: 'ChargeFilterValue', source: :values
+  end
+
+  class BillableMetric < ApplicationRecord
+    include Discard::Model
+    self.discard_column = :deleted_at
+    default_scope -> { kept }
+
+    has_many :groups, dependent: :delete_all
+    has_many :filters, -> { order(:key) }, dependent: :delete_all, class_name: 'BillableMetricFilter'
+  end
+
+  class ChargeFilter < ApplicationRecord
+    include Discard::Model
+    self.discard_column = :deleted_at
+    default_scope -> { kept }
+
+    belongs_to :charge
+    has_many :values, class_name: 'ChargeFilterValue', dependent: :destroy
+  end
+
+  def up
+    # NOTE: For each group, we create a filter with the same key and values
+    Group.find_each do |group|
+      filter = ::BillableMetricFilter.find_or_initialize_by(
+        billable_metric_id: group.billable_metric_id,
+        key: group.key,
+      )
+
+      filter.values ||= []
+      filter.values << group.value unless filter.values.include?(group.value)
```

Comment:
Performance/InefficientHashSearch: Use `#value?` instead of `#values.include?`. (https://github.com/fastruby/fast-ruby#hashkey-instead-of-hashkeysinclude-code)